business/namespaces.go

```diff
@@ -241,44 +327,19 @@ func (in *NamespaceService) GetNamespaces(ctx context.Context) ([]models.Namespa
 					k8sNamespaces = append(k8sNamespaces, *k8sNs)
 				}
 			}
-			namespaces = models.CastNamespaceCollection(k8sNamespaces)
+			namespaces = models.CastNamespaceCollection(k8sNamespaces, cluster)
 		}
 	}
 
 	result := namespaces
 
-	// exclude namespaces that are:
-	// 1. to be filtered out via the exclude list
-	// 2. to be filtered out via the label selector
-	// Note that the control plane namespace is never excluded
-	excludes := configObject.API.Namespaces.Exclude
-	if len(excludes) > 0 || labelSelectorExclude != "" {
-		result = []models.Namespace{}
-	NAMESPACES:
-		for _, namespace := range namespaces {
-			if namespace.Name != configObject.IstioNamespace {
-				if len(excludes) > 0 {
-					for _, excludePattern := range excludes {
-						if match, _ := regexp.MatchString(excludePattern, namespace.Name); match {
-							continue NAMESPACES
-						}
-					}
-				}
-				if labelSelectorExclude != "" {
-					if namespace.Labels[labelSelectorExcludeName] == labelSelectorExcludeValue {
-						continue NAMESPACES
-					}
-				}
-			}
-			result = append(result, namespace)
-		}
-	}
-
 	if kialiCache != nil {
-		kialiCache.SetNamespaces(in.k8s.GetToken(), result)
+		// Set namespace by user token
+		kialiCache.SetNamespaces(in.userClients[kubernetes.HomeClusterName].GetToken(), result)
```

Comment:
see my previous comment - i'm not sure tokens are the same across all clusters. Hardcoding to the home cluster might not work.

Reply:
I believe tokens are the same because all clusters are using the same OIDC provider. One token for all. @nrfox am I right?.